pkg/gatewayserver/grpc.go

```diff
@@ -48,3 +48,60 @@ func (gs *GatewayServer) GetGatewayConnectionStats(ctx context.Context, ids *ttn
 	stats, _ := val.(connectionEntry).Stats()
 	return stats, nil
 }
+
+func applyGatewayConnectionStatsFieldMask(
+	dst, src *ttnpb.GatewayConnectionStats,
+	paths ...string,
+) (*ttnpb.GatewayConnectionStats, error) {
+	if dst == nil {
+		dst = &ttnpb.GatewayConnectionStats{}
+	}
+	return dst, dst.SetFields(src, paths...)
+}
+
+// BatchGetGatewayConnectionStats gets statistics about gateway connections to the Gateway Server
+// of a batch of gateways.
+// This RPC skips unconnected gateways.
+// FieldMask paths can be used directly since they are sanitized by the middleware.
+func (gs *GatewayServer) BatchGetGatewayConnectionStats(
+	ctx context.Context,
+	req *ttnpb.BatchGetGatewayConnectionStatsRequest,
+) (*ttnpb.BatchGetGatewayConnectionStatsResponse, error) {
+	for _, ids := range req.GatewayIds {
+		if err := gs.entityRegistry.AssertGatewayRights(ctx, ids, ttnpb.Right_RIGHT_GATEWAY_STATUS_READ); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Returning an error at the first encounter of a gateway without `STATUS_READ` right forces the client into a process of trial and error until he has the right of all the requested gateways. I imagine this wouldn't be a problem for the console but might be for the CLI user.

Wouldn't it be best to have a list of gtws in which he doesn't have the right to read the stats?

Reply:
This will not scale. If a user makes a request with 100 gateway IDs and  doesn't have the rights to any of them, we will make 100 API calls to the IS and then return that to the caller.

Reply:
I do think it would be a good idea (especially if we want more bulk operations in the future) to also implement a bulk rights check.

Reply:
Yes bulk rights is the preferred way and I believe we have an issue for that somewhere. The implementation is out of scope for this PR.